UnitTests/CommonTestUtils/TempFileDeleter.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+
+namespace CommonTestUtils
+{
+    public sealed class TempFileDeleter : IDisposable
```

Comment:
Why not use something like https://docs.microsoft.com/en-us/dotnet/api/system.codedom.compiler.tempfilecollection?view=net-5.0

Reply:
I think this will suffice https://github.com/dotnet/winforms/blob/main/src/System.Windows.Forms/tests/TestUtilities/TempFile.cs

Reply:
Thank you, I searched myself - but too focused on tests. `TempFileCollection` is the almost perfect match.